Add basic docs for Metrics api and sdk (#2452)
open-telemetry/opentelemetry-dotnet: docs/metrics/customizing-the-sdk/README.md

As shown in the [getting-started](../getting-started/README.md) doc, a valid
[`MeterProvider`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#meterprovider)
must be configured and built to collect metrics with OpenTelemetry .NET Sdk.

> `MeterProvider` holds all the configuration for metrics like metricreaders,

Reviewer: Maybe use Pascal case? MetricReaders, Views, etc.

views, etc. Naturally, almost all the customizations must be done on the
`MeterProvider`.

## Building a MeterProvider

Building a `MeterProvider` is done using `MeterProviderBuilder` which must be
obtained by calling `Sdk.CreateMeterProviderBuilder()`. `MeterProviderBuilder`

> exposes various methods which configures the provider it is going to build.

Reviewer: configure*

> These includes methods like `AddSource`, `AddView` etc, and are explained in

Reviewer: include*

### Meter

[`Meter`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#meter)
is used for creating
[`Instruments`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument),
which are then used to report
[Measurements](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#measurement).
The SDK follows an explicit opt-in model for listening to meters. i.e, by
default, it listens to no meters. Every meter which is used to create
instruments must be explicitly added to the meter provider.

> `AddSource` method on `MeterProviderBuilder` can be used to add a `Meter` to the

Reviewer 2: Need to revisit all the changes in this PR to use AddMeter once the name change is merged.

> provider. The name of the `Meter` (case-insensitive) must be the argument to

Reviewer 2: ```suggestion
provider. The name of the `Meter` (case-insensitive) must be provided as an argument to
```

> this method. Multiple `AddSource` can be called to add more than one meter. It

Reviewer: How about changing this to:

> `AddSource` method can be called multiple times to add more than one meter.

> meter. It also supports wild-card subscription model.

Reviewer 2: ```suggestion
meters. It also supports wild-card subscription model.
```

> also supports wild-card subscription model as well.

Reviewer: Either have "also" or "as well".

It is not possible to add meters *after* the provider is built, by calling the
`Build()` method on the `MeterProviderBuilder`.

The snippet below shows how to add meters to the provider.

```csharp
using OpenTelemetry;
using OpenTelemetry.Metrics;

using var meterProvider = Sdk.CreateMeterProviderBuilder()
    // The following enables instruments from Meter
    // named "MyCompany.MyProduct.MyLibrary" only.
    .AddSource("MyCompany.MyProduct.MyLibrary")
    // The following enables instruments from all Meters
    // whose name starts with  "AbcCompany.XyzProduct.".
    .AddSource("AbcCompany.XyzProduct.*")
    .Build();
```

See [Program.cs](./Program.cs) for complete example.

> **Note**

Reviewer: Add colon after "Note".

A common mistake while configuring `MeterProvider` is forgetting to add all

> `Meter`s to the provider. It is recommended to leverage the wild card

Reviewer 2: ```suggestion
`Meter`s to the provider. It is recommended to leverage the wildcard
```
